controllers/binding/snapshotenvironmentbinding_controller_test.go

```diff
@@ -298,11 +298,19 @@ var _ = Describe("BindingController", func() {
 	})
 
 	It("can fail when Reconcile fails to prepare the adapter when Application is not found", func() {
-		Expect(k8sClient.Delete(ctx, hasApp)).Should(Succeed())
-		Eventually(func() error {
-			_, err := bindingReconciler.Reconcile(ctx, req)
-			return err
-		}).ShouldNot(BeNil())
+		err := k8sClient.Delete(ctx, hasApp)
+		Eventually(func() bool {
+			err := k8sClient.Get(ctx, types.NamespacedName{
+				Namespace: hasComp.ObjectMeta.Namespace,
+				Name:      hasComp.ObjectMeta.Name,
+			}, hasApp)
+			return err != nil
```

Comment:
Should we also check for `errors.IsNotFound(err))` here already to make this more robust?

Reply:
Fixed here and in #335